tests/unit/test_index_html_page.py

```diff
@@ -118,6 +118,34 @@ def test_get_html_response_no_head(url):
     ]
 
 
+def test_get_html_response_dont_log_clear_text_password(caplog):
+    """
+    `_get_html_response()` shouldn't send a HEAD request if the URL does not
+    look like an archive, only the GET request that retrieves data.
+    """
+    password = "my_password"
+    url_template = "https://user:{}@example.com/simple/"
+    session = mock.Mock(PipSession)
+
+    # Mock the headers dict to ensure it is accessed.
+    session.get.return_value = mock.Mock(headers=mock.Mock(**{
+        "get.return_value": "text/html",
+    }))
+
+    caplog.set_level(logging.DEBUG)
+
+    resp = _get_html_response(url_template.format(password), session=session)
```

Comment:
Instead of using `url_template`, it would be simpler and more direct just to pass `https://user:my_password@example.com/simple/`.